packages/share_plus/share_plus_platform_interface/lib/method_channel/method_channel_share.dart

```diff
@@ -141,19 +144,56 @@ class MethodChannelShare extends SharePlatform {
     String? subject,
     String? text,
     Rect? sharePositionOrigin,
-  }) {
-    final mimeTypes =
-        files.map((e) => e.mimeType ?? _mimeTypeForPath(e.path)).toList();
+  }) async {
+    final filesWithPath = await _getFiles(files);
+
+    final mimeTypes = filesWithPath
+        .map((e) => e.mimeType ?? _mimeTypeForPath(e.path))
+        .toList();
 
     return shareFilesWithResult(
-      files.map((e) => e.path).toList(),
+      filesWithPath.map((e) => e.path).toList(),
       mimeTypes: mimeTypes,
       subject: subject,
       text: text,
       sharePositionOrigin: sharePositionOrigin,
     );
   }
 
+  /// if file doesn't contain path
+  /// then make new file in TemporaryDirectory and return with path
+  ///
+  /// the system will automatically delete files in this
+  /// TemporaryDirectory as disk space is needed elsewhere on the device
+  Future<List<XFile>> _getFiles(List<XFile> files) async {
+    int count = 1;
+
+    if (!files.every((element) => element.path.isNotEmpty)) {
+      final newFiles = <XFile>[];
+
+      final String tempPath = (await getTemporaryDirectory()).path;
+
+      for (final XFile element in files) {
+        if (element.path.isEmpty) {
+          final path = '$tempPath/${element.name}$count';
```

Comment:
I've been testing this and seems that the filename is not correctly preserved, in my test I create the XFile and set the name as `name: 'flutter_logo.png',` however on share I only see a `1` being shared.

Is there a way this can be improved?

I have modified the example code to showcase this.

Reply:
Actually, the `name` parameter doesn't work as expected, see:

Reply:
its because Xfile is from cross_file and passed name is not assigned anywhere inside cross file lib,

so in above code element.name is null

Reply:
I think a better solution would be to generate a unique name based on a random uuid (you can use the uuid package) and a file extension based off the provided mime type.

Reply:
i would go unique name based on a random uuid.